core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1786,6 +1786,105 @@ object TestUtils extends Logging {
       KafkaYammerMetrics.defaultRegistry.removeMetric(metricName)
   }
 
+  def clearYammerMetricsExcept(names: Set[String]): Unit = {
```

Comment:
This seems like a pretty heavy hammer that might cause problems, given that so many tests modify global metrics state. Why do we need this now, when we didn't before?